server/dataone_package.py

```diff
@@ -93,63 +114,179 @@ def create_minimum_eml(tale, user, item_ids, eml_pid):
+        :return:
+        """
         other_entity = ET.SubElement(dataset, 'otherEntity')
-        ET.SubElement(other_entity, 'entityName').text = item['name']
-        ET.SubElement(other_entity, 'entityDescription').text = description if\
+        ET.SubElement(other_entity, 'entityName').text = name
+        ET.SubElement(other_entity, 'entityDescription').text = description if \
             bool(len(description)) else 'None'
-
-        physical = ET.SubElement(other_entity, 'physical')
-        ET.SubElement(physical, 'objectName').text = item['name']
+        return other_entity
+
+    def create_physical(other_entity_section, name, size):
+        """
+        Creates a `physical` section.
+        :param other_entity_section: The super-section
+        :param name: The name of the object
+        :param size: The size in bytes of the object
+        :type other_entity_section: xml.etree.ElementTree.Element
+        :type name: str
+        :type size: str
+        :return: The physical section
+        :rtype: xml.etree.ElementTree.Element
+        """
+        physical = ET.SubElement(other_entity_section, 'physical')
+        ET.SubElement(physical, 'objectName').text = name
         size_element = ET.SubElement(physical, 'size')
-        size_element.text = str(item['size'])
+        size_element.text = str(size)
         size_element.set('unit', 'bytes')
-        format = get_file_format(item_id, user)
-        data_format = ET.SubElement(physical, 'dataFormat')
+        return physical
+
+    def create_format(format, physical_section):
+        data_format = ET.SubElement(physical_section, 'dataFormat')
         externally_defined = ET.SubElement(data_format, 'externallyDefinedFormat')
         ET.SubElement(externally_defined, 'formatName').text = format
 
-        ET.SubElement(other_entity, 'entityType').text = 'Other'
+    def add_object_record(name, description, size, format):
+        """
+        Add a section to the EML that describes an object.
+        :param name: The name of the object
+        :param description: The object's description
+        :param size: The size of the object
+        :param format: The format type
+        :return: None
+        """
+        other_entity_section = create_other_entity(name, description)
+        physical_section = create_physical(other_entity_section,
+                                           name,
+                                           size)
+        create_format(format, physical_section)
+        ET.SubElement(other_entity_section, 'entityType').text = 'Other'
+
+    # Add a <otherEntity> block for each object
+    for item_id in item_ids:
+        item = Item().load(item_id, user=user)
+        format = get_file_format(item_id, user)
+
+        # Create the record for the object
+        add_object_record(item['name'], item['description'], item['size'], format)
+
+    # Add a section for the file describing any remote objects
+    if bool(external_data):
```

Comment:
Is `bool(Dict)` more Pythonic than just `len(Dict)` to test emptiness?

Reply:
I'm not sure if the community prefers one way or another,
https://stackoverflow.com/questions/23177439/python-checking-if-a-dictionary-is-empty-doesnt-seem-to-work

Reply:
Oh boy. Gotta love Python. I'm fine either way.

Reply:
Plus this already exists in the codebase:

```python
❯ ag "bool\("
server/rest/tale.py
159:                iframe=bool(tale.get('iframe', False))
```